include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,326 @@
+    requires std::Same<reference_t<range_type>, cigar<> &>
+//!\endcond
+std::ostream & operator<<(std::ostream & os, range_type const & cigv)
+{
+    std::for_each(cigv.begin(), cigv.end(), [&os] (auto cig) { os << cig; } );
+    return os;
+}
+
+} // namespace seqan3
+
+// ------------------------------------------------------------------
+// create cigar_vector out of alignment
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+/*!\brief Compares two aligned sequence values and returns their seqan3::cigar_op.
+ * \ingroup aligned_sequence
+ * \tparam reference_value_type Must be equality comparable to seqan3::gap.
+ * \tparam query_value_type     Must be equality comparable to seqan3::gap.
+ * \param  reference_value      Considered indirectly related to the cigar operation.
+ * \param  query_value          The value the cigar operation is based on.
+ * \param  extended_cigar Whether to print the extended cigar alphabet or not. See seqan3::cigar_op.
+ * \returns A seqan3::cigar_op representing the alignment operation between the
+ *          two values.
+ *
+ * \details
+ *
+ * \attention Note that a seqan3::cigar_op, of a cigar element describing an
+ *            alignment at a certain position, is always based on one of the
+ *            two sequences in the pairwise alignment. In this case, the resulting
+ *            operation is based on the \p query value.
+ */
+template<typename reference_value_type, typename query_value_type>
+//!\cond
+    requires std::EqualityComparableWith<std::remove_reference_t<reference_value_type>, gap> &&
+             std::EqualityComparableWith<std::remove_reference_t<query_value_type>, gap>
+//!\endcond
+cigar_op compare_aligned_values(reference_value_type const reference_value,
+                                query_value_type const query_value,
+                                bool const extended_cigar)
+{
+    return (reference_value == gap::GAP)
+        ? (query_value == gap::GAP)
+            ? cigar_op::P
+            : cigar_op::I
+        : (query_value == gap::GAP)
+            ? cigar_op::D
+            : (extended_cigar)
+                ? (query_value == reference_value)
+                    ? cigar_op::EQ
+                    : cigar_op::X
+                : cigar_op::M;
+}
+
+/*!\brief Compares two aligned sequence values and returns their seqan3::cigar_op.
+ * \ingroup aligned_sequence
+ * \tparam ref_seq_type   Must model seqan3::aligned_sequence_concept.
+ * \tparam query_seq_type Must model seqan3::aligned_sequence_concept.
+ * \param  alignment      The alignment, represented by a pair of aligned sequences,
+ *                        to be transformed into seqan3::cigar_vector based on the
+ *                        second (query) sequence.
+ * \param  offset_start   The offset at the start of the query sequence indicating soft-clipping.
+ * \param  offset_end     The offset at the end of the query sequence indicating soft-clipping.
```

Comment:
offset is not a good expression at the end of a sequence. 
Think about sth like `std::pair<uint,uint> aligned_region` or `uint align_start` + `align_end`...